modules/plugin/jdbc/jdbc-hana/src/main/java/org/geotools/data/hana/HanaDialect.java

```diff
@@ -394,6 +409,58 @@ public CoordinateReferenceSystem createCRS(int srid, Connection cx) throws SQLEx
         }
     }
 
+    @Override
+    public List<ReferencedEnvelope> getOptimizedBounds(String schema, SimpleFeatureType featureType, Connection cx) throws SQLException, IOException {
+        if (!isEstimatedExtentsEnabled())
+            return super.getOptimizedBounds(schema, featureType, cx);
+
+        String tableName = featureType.getTypeName();
+        if (dataStore.getVirtualTables().get(tableName) != null) {
+            return null;
+        }
+
+        List<ReferencedEnvelope> result = new ArrayList<>();
+        PreparedStatement ps = null;
+        ResultSet rs = null;
+        schema = resolveSchema(schema, cx);
+        try {
+            for (AttributeDescriptor att : featureType.getAttributeDescriptors()) {
+                if (att instanceof GeometryDescriptor) {
+                    StringBuffer sql = new StringBuffer();
+                    sql.append("SELECT MIN_X, MAX_X, MIN_Y, MAX_Y FROM SYS.M_ST_GEOMETRY_COLUMNS WHERE SCHEMA_NAME=? AND TABLE_NAME=? AND COLUMN_NAME=?");
+
+                    ps = cx.prepareStatement(sql.toString());
+                    ps.setString(1, schema);
+                    ps.setString(2, tableName);
+                    ps.setString(3, att.getName().getLocalPart());
+
+                    rs = ps.executeQuery();
+                    if (rs.next()) {
+                        double xMin = rs.getDouble(1);
+                        double xMax = rs.getDouble(2);
+                        double yMin = rs.getDouble(3);
+                        double yMax = rs.getDouble(4);
+
+                        // Row for column is available and data is present
+                        if (!rs.wasNull())
```

Comment:
If we cannot determine the extent, sometimes we return null and sometimes we return an empty list. I think, we should always return null in those cases.

Reply:
I adjusted the logic a bit. As soon as one of the rows contains NULL entries, we return NULL to indicate a failure. If the list is empty e.g. if there was no row for the column in the first place, we return NULL as well.

Reply:
If multiple column were requested, the same thing should happen. As soon as one of the column does not have a valid extent in the table, we return NULL.